appvm-scripts/usrbin/qubes-run-xorg.sh

```diff
@@ -69,4 +69,11 @@ chmod 770 /var/run/xf86-qubes-socket
 
 export XDG_SEAT=seat0 XDG_VTNR=7 XDG_SESSION_CLASS=user
 
-exec su -l user -c "/usr/bin/xinit $XSESSION -- $XORG :0 -nolisten tcp vt07 -wr -config xorg-qubes.conf > ~/.xsession-errors 2>&1"
+# Defaults value in case default-user value is not available
+if [ ! -z "$(qubesdb-read /default-user)" ];then
+    DEFAULT_USER=$(qubesdb-read /default-user)
```

Comment:
Negation of `-z` is `-n`.
Better save the value to some variable and use it instead of reading from qubesdb twice (what if only the second read fails?). You can even use `DEFAULT_USER` for that, and later set it to `user`, if its empty.